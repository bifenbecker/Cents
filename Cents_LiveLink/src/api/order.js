import httpClient from "./httpClient";

export const fetchStoreSettings = params => {
  return httpClient({
    method: "GET",
    url: "/live-status/settings",
    params,
  });
};

export const authenticateOrder = token => {
  return httpClient({
    method: "GET",
    url: "/live-status/verify-order",
    params: {
      token,
    },
  });
};

export const fetchOrderDetail = token => {
  return httpClient({
    method: "GET",
    url: "/live-status",
    params: {
      token,
    },
  });
};

export const updateTip = (token, tipData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/tip",
    params: {
      token,
    },
    data: tipData,
  });
};

export const addPromo = (token, promoData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-promotion",
    params: {
      token,
    },
    data: promoData,
  });
};

export const removePromo = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-promotion",
    params: {
      token,
    },
  });
};

export const addCredits = (token, creditsData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-credits",
    params: {
      token,
    },
    data: creditsData,
  });
};

export const removeCredits = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-credits",
    params: {
      token,
    },
  });
};

export const setOnlineOrderReturnMethod = (token, data) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/return-method",
    params: {
      token,
    },
    data,
  });
};

export const updatePaymentMethodIntent = (token, {paymentToken}) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/payment-method",
    params: {
      token,
    },
    data: {paymentToken},
  });
};

export const cancelOrder = params => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/order/cancel",
    params,
  });
};

export const getRouteDelivery = (token, {orderDelivery}) => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-delivery/${orderDelivery}/route-details`,
    params: {token},
  });
};

export const fetchOrderDeliveries = token => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-deliveries`,
    params: {token},
  });
};

export const manageOrder = (token, data) => {
  return httpClient({
    method: "PUT",
    url: `/live-status/live-link/manage`,
    params: {token},
    data,
  });
};
