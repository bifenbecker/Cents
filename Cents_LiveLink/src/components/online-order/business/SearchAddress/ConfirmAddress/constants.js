export const PLACE_HOLDER = {
  ADDRESS_DETAILS: "Address details",
  PICKUP_INSTRUCTIONS:
    "e.g. call upon arrival, entrance on right side of building, don’t ring bell, etc.",
};

export const LABEL_TEXT = {
  ADDRESS_DETAILS: "Apt / Suite / Unit",
  PICKUP_INSTRUCTIONS: "Pickup Instructions",
};

export const SUBTITLE_TEXT = {
  ADDRESS_DETAILS: "Confirm your address for pickup",
  PICKUP_INSTRUCTIONS: "Pickup Instructions",
};

export const CHECKBOX_LABEL = "Leave it at my door if I’m not home";

export const GET_BTN_TEXT = (flag) => {
  return flag ? "Save & Continue" : "Save";
};

export const NOTES_TEXT = "We will send you updates about your pickup via text.";
