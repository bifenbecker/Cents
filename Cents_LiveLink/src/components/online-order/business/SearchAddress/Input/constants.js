export const SEARCH_FOR_A_NEW_ADDRESS = "Search for a new address";
export const SEARCH_FOR_AN_ADDRESS = "Search for an address";
export const SEARCH_ADDRESS = "Search Address";
