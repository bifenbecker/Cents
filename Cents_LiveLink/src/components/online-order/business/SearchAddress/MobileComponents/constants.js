export const STATES = {
  CONFIRM: "CONFIRM",
  CANT_PICKUP: "CANT_PICKUP",
  AUTO_COMPLETE: "AUTO_COMPLETE",
  SAVED_ADDRESSES: "SAVED_ADDRESSES",
};
