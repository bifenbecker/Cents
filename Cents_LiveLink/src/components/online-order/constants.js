export const orderPickUpSteps = {
  details: "order-details",
  finishingUp: "finishing-up",
};

export const deliveryProviders = {
  uber: "UBER",
  ownDriver: "OWN_DRIVER",
  doorDash: "DOORDASH",
};

export const orderFields = [
  "servicePriceId",
  "serviceModifierIds",
  "customerNotes",
  "isHangDrySelected",
  "hangDryInstructions",
  "orderNotes",
  "customerAddressId",
  "paymentToken",
  "promoCode",
  "bagCount",
  "returnMethod",
  "hasDryCleaning",
];

export const mandatoryOrderFields = [
  "paymentToken",
  "returnMethod",
  "servicePriceId",
  "customerAddressId",
];

export const bufferRequiredForOrder = {
  [deliveryProviders.uber]: 10,
  [deliveryProviders.doorDash]: 10,
  [deliveryProviders.ownDriver]: 20,
};

export const orderDeliveryFields = ["pickup", "delivery"];

export const orderDeliverySubFields = [
  "type",
  "timingsId",
  "deliveryProvider",
  "deliveryWindow",
  "thirdPartyDeliveryId",
  "totalDeliveryCost",
  "courierTip",
  "subsidyInCents",
  "thirdPartyDeliveryCostInCents",
];

export const onDemandDeliveryTypes = {
  pickup: "PICKUP",
  delivery: "DELIVERY",
  pickupAndDelivery: "PICKUP_AND_DELIVERY",
};

export const SERVICE_CATEGORY_TYPES = {
  DRY_CLEANING: "DRY_CLEANING",
  LAUNDRY: "LAUNDRY",
  ALTERATIONS: "ALTERATIONS",
};

export const cloneOrderState = {
  REVIEW: "REVIEW_CLONE_ORDER",
  EDIT: "EDIT_CLONE_ORDER",
  INITIAL: "NO_CLONE_ORDER",
};
