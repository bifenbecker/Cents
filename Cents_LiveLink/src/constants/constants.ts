export const OTP_CODE_LENGTH = 6;
export const PHONE_LENGTH = 10;

export const PHONE_NUMBER_FORMAT_REGEXP = /^(\d{3})(\d{3})(\d{4})$/;
export const NOT_DIGITS_REGEXP = /\D/g;
