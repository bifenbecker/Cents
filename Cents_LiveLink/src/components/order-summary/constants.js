import {RETURN_METHODS} from "../../constants/order";
import {
  CALL_ICON,
  DELIVERY_ORDER_COMPLETED,
  ON_DEMAND_DELIVERY_IN_PROGRESS,
  ON_DEMAND_PICKUP_COMPLETED,
  ON_DEMAND_PICKUP_INITIATED,
  ON_DEMAND_PICKUP_SCHEDULED,
  ORDER_CANCELED,
  ORDER_COMPLETED,
  PROCESSING,
  READY_FOR_CUSTOMER_PICKUP,
  SMS_ICON,
  STD_DELIVERY_IN_PROGRESS,
  STD_PICKUP_COMPLETED,
  STD_PICKUP_INITIATED,
  STD_PICKUP_SCHEDULED,
} from "../../assets/images";

export {RETURN_METHODS, ORDER_TYPES} from "../../constants/order";

export const RESIDENTIAL_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Received & Processing",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const ONLINE_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Received & Processing",
  readyForPickup: "Laundry Ready",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const SERVICE_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Processing",
  readyForPickup: "Laundry Ready",
  completed: "Complete",
};

export const DELIVERY_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Processing",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const TIMELINE_CANCELED_STATUS = "Canceled";

export const DELIVERY_TRACKING_ORDER_STATUSES = {
  submitted: "SUBMITTED",
  enRouteToPickup: "EN_ROUTE_TO_PICKUP",
  enRouteToDropoff: "EN_ROUTE_TO_DROPFF",
  completed: "COMPLETED",
};

export const RETURN_METHODS_DISPLAY = {
  [RETURN_METHODS.inStorePickup]: "NO THANKS",
  [RETURN_METHODS.delivery]: "DELIVER TO ME",
};

export const serviceSelections = {
  laundry: "Laundry",
  dryCleaning: "Dry Cleaning",
};

export const orderItemCategories = {
  deliveryOrPickup: "DELIVERY",
  fixedPriceItem: "FIXED_PRICE",
  perPoundItem: "PER_POUND",
};

export const ORDER_STATUSES = {
  COMPLETED: "COMPLETED",
  CANCELLED: "CANCELLED",
};

export const IMAGE_IDS = {
  DELIVERY_ORDER_COMPLETED: DELIVERY_ORDER_COMPLETED,
  ON_DEMAND_DELIVERY_IN_PROGRESS: ON_DEMAND_DELIVERY_IN_PROGRESS,
  ON_DEMAND_PICKUP_COMPLETED: ON_DEMAND_PICKUP_COMPLETED,
  ON_DEMAND_PICKUP_INITIATED: ON_DEMAND_PICKUP_INITIATED,
  ON_DEMAND_PICKUP_SCHEDULED: ON_DEMAND_PICKUP_SCHEDULED,
  ORDER_CANCELED: ORDER_CANCELED,
  ORDER_COMPLETED: ORDER_COMPLETED,
  PROCESSING: PROCESSING,
  READY_FOR_CUSTOMER_PICKUP: READY_FOR_CUSTOMER_PICKUP,
  STD_DELIVERY_IN_PROGRESS: STD_DELIVERY_IN_PROGRESS,
  STD_PICKUP_COMPLETED: STD_PICKUP_COMPLETED,
  STD_PICKUP_SCHEDULED: STD_PICKUP_SCHEDULED,
  STD_PICKUP_INITIATED: STD_PICKUP_INITIATED,
  CALL_ICON,
  SMS_ICON,
};

export const PAYMENT_STATUSES = {
  balanceDue: "BALANCE_DUE",
  paid: "PAID",
  pending: "PENDING",
};
