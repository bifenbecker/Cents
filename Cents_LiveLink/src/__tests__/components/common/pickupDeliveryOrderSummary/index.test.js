import React from "react";
import {render} from "@testing-library/react";

import PickupDeliveryOrderSummary from "../../../../components/common/pickupDeliveryOrderSummary";
import {DELIVERY_PROVIDERS} from "../../../../constants/order";

const renderPickupDeliveryOrderSummary = props => {
  const {getByText, container} = render(<PickupDeliveryOrderSummary {...props} />);
  return {getByText, container};
};

describe("Testing pickup and delivery order summary Component", () => {
  describe("Should render relevant child components based on combinations of pickup and delivery providers", () => {
    it("Case - Pickup - No, Delivery - Own Driver", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
      });
      expect(getByText("Delivery")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it("Case - Pickup - No, Delivery - Not Own Driver", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
      });
      expect(getByText("On-Demand Delivery")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it("Case - Pickup - Own Driver, Delivery - Own Driver", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
      });
      expect(getByText("Pickup & Delivery")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it("Case - Pickup - Own Driver, Delivery - No", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
      });
      expect(getByText("Pickup")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it("Case - Pickup - Not Own Driver, Delivery - No", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
      });
      expect(getByText("On-Demand Pickup")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it("Case - Pickup - Not Own Driver, Delivery - Own Driver", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
      });
      expect(getByText("On-Demand Pickup")).toBeInTheDocument();
      expect(getByText("Delivery")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it("Case - Pickup - Own Driver, Delivery - Not Own Driver", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.uber},
      });
      expect(getByText("Pickup")).toBeInTheDocument();
      expect(getByText("On-Demand Delivery")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it("Case - Pickup - Not Own Driver, Delivery - Not Own Driver", () => {
      const {getByText, container} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.uber},
      });
      expect(getByText("On-Demand Pickup & Delivery")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });

  describe("On demand summary - should calculate tip and show in different cases", () => {
    it("Case - tip provided for both pickup and delivery", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        pickupDeliveryTip: 1,
        returnDeliveryTip: 1,
      });
      expect(getByText("Split evenly between both drivers")).toBeInTheDocument();
      expect(getByText("Pickup & Delivery Driver Tips")).toBeInTheDocument();
      expect(getByText("$2.00")).toBeInTheDocument();
    });

    it("Case - tip provided for pickup only", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        pickupDeliveryTip: 4,
      });
      expect(getByText("Pickup Driver Tip")).toBeInTheDocument();
      expect(getByText("$4.00")).toBeInTheDocument();
    });

    it("Case - tip provided for delivery only", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        returnDeliveryTip: 6,
      });
      expect(getByText("Delivery Driver Tip")).toBeInTheDocument();
      expect(getByText("$6.00")).toBeInTheDocument();
    });

    it("Case - own driver pickup, on demand delivery tip provided", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        returnDeliveryTip: 6,
      });
      expect(getByText("Delivery Driver Tip")).toBeInTheDocument();
      expect(getByText("$6.00")).toBeInTheDocument();
    });

    it("Case - own driver delivery, on demand pickup tip provided", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
        returnDeliveryTip: 6,
      });
      expect(getByText("Delivery Driver Tip")).toBeInTheDocument();
      expect(getByText("$6.00")).toBeInTheDocument();
    });
  });

  describe("Standard Summary - should show pickup and delivery fees in different cases", () => {
    it("Pickup and delivery - should show pickup and delivery fees", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        pickupDeliveryFee: 6,
        returnDeliveryFee: 2,
      });
      expect(getByText("($4.00 each way)")).toBeInTheDocument();
    });

    it("Pickup only - should show FREE if no pickup fee", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        pickupDeliveryFee: 7,
      });
      expect(getByText("$7.00")).toBeInTheDocument();
    });

    it("Pickup only - should show pickup fee", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
      });
      expect(getByText("FREE")).toBeInTheDocument();
    });

    it("Delivery only - should show FREE if no delivery fee", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
        returnDeliveryFee: 5,
      });
      expect(getByText("$5.00")).toBeInTheDocument();
    });

    it("Delivery only - should show delivery fee", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.ownDriver},
      });
      expect(getByText("FREE")).toBeInTheDocument();
    });
  });

  describe("On demand Summary - should show pickup and delivery fees in different cases", () => {
    it("Should calculate and display trip cost", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        delivery: {
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
          totalDeliveryCost: 4,
          subsidyInCents: 400,
          thirdPartyDeliveryCostInCents: 400,
        },
      });
      expect(getByText("Trip cost: $8.00")).toBeInTheDocument();
    });

    it("Should show laundry subsidy correctly", () => {
      let getByText = renderPickupDeliveryOrderSummary({
        delivery: {
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
          totalDeliveryCost: 4,
          subsidyInCents: 400,
          thirdPartyDeliveryCostInCents: 200,
        },
      }).getByText;
      expect(getByText("Subsidy: -$2.00")).toBeInTheDocument();

      getByText = renderPickupDeliveryOrderSummary({
        delivery: {
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
          totalDeliveryCost: 4,
          subsidyInCents: 300,
          thirdPartyDeliveryCostInCents: 600,
        },
      }).getByText;
      expect(getByText("Subsidy: -$3.00")).toBeInTheDocument();
    });

    it("Should show driver fee if store location is in California", () => {
      const {getByText} = renderPickupDeliveryOrderSummary({
        delivery: {
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
        },
        storeLocation: "CA",
      });
      expect(getByText("Includes CA Driver Fee: $2.00")).toBeInTheDocument();
    });

    it("Should show on demand pickup and delivery fees correctly", () => {
      const orderDetails = {
        pickup: {
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
        },
        delivery: {
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
        },
      };
      let getByText = renderPickupDeliveryOrderSummary({
        ...orderDetails,
        pickupDeliveryFee: 2,
        returnDeliveryFee: 3,
        storeLocation: "California",
      }).getByText;
      expect(getByText("(~ $2.50 each way)")).toBeInTheDocument();
      expect(
        getByText("Includes CA Driver Fee: $4.00 ($2.00 each way)")
      ).toBeInTheDocument();

      getByText = renderPickupDeliveryOrderSummary({
        ...orderDetails,
        pickupDeliveryFee: 2,
      }).getByText;
      expect(getByText("(pickup: $2.00, delivery: $0.00)")).toBeInTheDocument();

      getByText = renderPickupDeliveryOrderSummary({
        ...orderDetails,
        returnDeliveryFee: 3,
      }).getByText;
      expect(getByText("(pickup: $0.00, delivery: $3.00)")).toBeInTheDocument();
    });

    it("Should show on demand pickup and delivery subsidy correctly", () => {
      const orderDetails = {
        pickup: {
          subsidyInCents: 200,
          thirdPartyDeliveryCostInCents: 100,
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
        },
        delivery: {
          subsidyInCents: 300,
          thirdPartyDeliveryCostInCents: 200,
          deliveryProvider: DELIVERY_PROVIDERS.doorDash,
        },
      };
      let getByText = renderPickupDeliveryOrderSummary({
        ...orderDetails,
      }).getByText;
      expect(getByText("Subsidy: -$3.00")).toBeInTheDocument();
      expect(getByText("(-$1.50 each way)")).toBeInTheDocument();

      getByText = renderPickupDeliveryOrderSummary({
        pickup: orderDetails.pickup,
        delivery: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
      }).getByText;
      expect(getByText("Subsidy: -$1.00")).toBeInTheDocument();
      expect(getByText("(pickup: -$1.00, delivery: $0.00)")).toBeInTheDocument();

      getByText = renderPickupDeliveryOrderSummary({
        delivery: orderDetails.delivery,
        pickup: {deliveryProvider: DELIVERY_PROVIDERS.doorDash},
      }).getByText;
      expect(getByText("Subsidy: -$2.00")).toBeInTheDocument();
      expect(getByText("(pickup: $0.00, delivery: -$2.00)")).toBeInTheDocument();
    });
  });
});
