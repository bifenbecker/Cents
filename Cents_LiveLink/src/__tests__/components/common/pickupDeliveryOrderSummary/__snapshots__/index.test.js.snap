// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - No, Delivery - Not Own Driver 1`] = `
<div>
  <div
    class="sc-bczRLJ cYXLjH"
    width="100%"
  >
    <div
      class="sc-bczRLJ sc-gsnTZi kaLzzZ hzPtvw"
      color="TEXT_GREY"
      font-family="secondary"
      font-size="1,2"
      font-weight="normal"
      width="100%"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        On-Demand 
        Delivery
      </div>
      <div
        class="sc-bczRLJ jITePI"
      >
        $0.00
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - No, Delivery - Own Driver 1`] = `
<div>
  <div
    class="sc-bczRLJ sc-gsnTZi dyCyXN hzPtvw"
    color="TEXT_GREY"
    font-family="secondary"
    font-size="1,2"
    font-weight="normal"
    width="100%"
  >
    <div
      class="sc-bczRLJ sc-gsnTZi jITePI hzPtvw"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        Delivery
      </div>
    </div>
    <div
      class="sc-bczRLJ jITePI"
    >
      FREE
    </div>
  </div>
</div>
`;

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - Not Own Driver, Delivery - No 1`] = `
<div>
  <div
    class="sc-bczRLJ cYXLjH"
    width="100%"
  >
    <div
      class="sc-bczRLJ cYXLjH"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi kaLzzZ hzPtvw"
        color="TEXT_GREY"
        font-family="secondary"
        font-size="1,2"
        font-weight="normal"
        width="100%"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          On-Demand 
          Pickup
        </div>
        <div
          class="sc-bczRLJ jITePI"
        >
          $0.00
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - Not Own Driver, Delivery - Not Own Driver 1`] = `
<div>
  <div
    class="sc-bczRLJ cYXLjH"
    width="100%"
  >
    <div
      class="sc-bczRLJ sc-gsnTZi kaLzzZ hzPtvw"
      color="TEXT_GREY"
      font-family="secondary"
      font-size="1,2"
      font-weight="normal"
      width="100%"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        On-Demand Pickup & Delivery
      </div>
      <div
        class="sc-bczRLJ jITePI"
      >
        $0.00
      </div>
    </div>
    <div
      class="sc-bczRLJ sc-gsnTZi hasscr hzPtvw"
      color="TEXT_GREY"
      font-family="secondary"
      font-size="1,0"
      font-style="italic"
      font-weight="normal"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        Trip cost: 
        $0.00
      </div>
      <div
        class="sc-bczRLJ cTiMdd"
        font-style="italic"
      />
    </div>
  </div>
</div>
`;

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - Not Own Driver, Delivery - Own Driver 1`] = `
<div>
  <div
    class="sc-bczRLJ cYXLjH"
    width="100%"
  >
    <div
      class="sc-bczRLJ cYXLjH"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi kaLzzZ hzPtvw"
        color="TEXT_GREY"
        font-family="secondary"
        font-size="1,2"
        font-weight="normal"
        width="100%"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          On-Demand 
          Pickup
        </div>
        <div
          class="sc-bczRLJ jITePI"
        >
          $0.00
        </div>
      </div>
    </div>
    <div
      class="sc-bczRLJ sc-gsnTZi dyCyXN hzPtvw"
      color="TEXT_GREY"
      font-family="secondary"
      font-size="1,2"
      font-weight="normal"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi jITePI hzPtvw"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          Delivery
        </div>
      </div>
      <div
        class="sc-bczRLJ jITePI"
      >
        FREE
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - Own Driver, Delivery - No 1`] = `
<div>
  <div
    class="sc-bczRLJ sc-gsnTZi dyCyXN hzPtvw"
    color="TEXT_GREY"
    font-family="secondary"
    font-size="1,2"
    font-weight="normal"
    width="100%"
  >
    <div
      class="sc-bczRLJ sc-gsnTZi jITePI hzPtvw"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        Pickup
      </div>
    </div>
    <div
      class="sc-bczRLJ jITePI"
    >
      FREE
    </div>
  </div>
</div>
`;

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - Own Driver, Delivery - Not Own Driver 1`] = `
<div>
  <div
    class="sc-bczRLJ cYXLjH"
    width="100%"
  >
    <div
      class="sc-bczRLJ sc-gsnTZi dyCyXN hzPtvw"
      color="TEXT_GREY"
      font-family="secondary"
      font-size="1,2"
      font-weight="normal"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi jITePI hzPtvw"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          Pickup
        </div>
      </div>
      <div
        class="sc-bczRLJ jITePI"
      >
        FREE
      </div>
    </div>
    <div
      class="sc-bczRLJ cYXLjH"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi kaLzzZ hzPtvw"
        color="TEXT_GREY"
        font-family="secondary"
        font-size="1,2"
        font-weight="normal"
        width="100%"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          On-Demand 
          Delivery
        </div>
        <div
          class="sc-bczRLJ jITePI"
        >
          $0.00
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing pickup and delivery order summary Component Should render relevant child components based on combinations of pickup and delivery providers Case - Pickup - Own Driver, Delivery - Own Driver 1`] = `
<div>
  <div
    class="sc-bczRLJ sc-gsnTZi dyCyXN hzPtvw"
    color="TEXT_GREY"
    font-family="secondary"
    font-size="1,2"
    font-weight="normal"
    width="100%"
  >
    <div
      class="sc-bczRLJ sc-gsnTZi jITePI hzPtvw"
    >
      <div
        class="sc-bczRLJ jITePI"
      >
        Pickup & Delivery
      </div>
    </div>
    <div
      class="sc-bczRLJ jITePI"
    >
      FREE
    </div>
  </div>
</div>
`;
